Composite: improve Storybook examples and clean up prop docs

ciampo opened the pull request:
<!-- Thanks for contributing to Gutenberg! Please follow the Gutenberg Contributing Guidelines:
https://github.com/WordPress/gutenberg/blob/trunk/CONTRIBUTING.md -->

## What?
<!-- In a few words, what is the PR actually doing? -->

Follow-up to #63564

Improve Storybook examples for the `Composite` component.

## Why?
<!-- Why is this PR necessary? What problem is it solving? Reference any existing previous issue(s) or PR(s), but please add a short summary here, too -->

Better documentation for `Composite`

## How?
<!-- How is your PR addressing the issue at hand? What are the implementation details? -->

- add interactive controls
- add more examples on top of the existing grid example (default, groups)

## Testing Instructions
<!-- Please include step by step instructions on how to test this PR. -->
<!-- 1. Open a post or page. -->
<!-- 2. Insert a heading block. -->
<!-- 3. etc. -->

Open Storybook, go to the `Composite` component:

- make sure interactive controls behave as expected
- make sure the newly added examples make sense and offer a more complete showcase of the component's offering.


This PR doesn't include runtime changes to the actual component.

ciampo committed:
More clear tranform function comments

ciampo committed:
Add interactive controls

ciampo committed:
Add simpler default Storybook example

ciampo committed:
Improve `activeId`'s prop description

ciampo committed:
Add Groups example

ciampo committed:
CHANGELOG

mirka commented on packages/components/src/composite/stories/index.story.tsx:
```diff
@@ -29,7 +29,26 @@ const meta: Meta< typeof UseCompositeStorePlaceholder > = {
+		},
+	},
 	parameters: {
+		actions: { argTypesRegex: '^on.*' },
```
FYI #60660

I'm guessing we don't need actions at all in this case?

ciampo replied:
Wasn't aware of #60660, thank you for flagging. I will remove this line.

mirka commented on packages/components/src/composite/stories/utils.tsx:
Can we have a better description for this too? Suggestion:

```diff
diff --git a/packages/components/src/composite/stories/utils.tsx b/packages/components/src/composite/stories/utils.tsx
index febbffd00b..8ce0410bf7 100644
--- a/packages/components/src/composite/stories/utils.tsx
+++ b/packages/components/src/composite/stories/utils.tsx
@@ -9,7 +9,9 @@ import type { StoryContext } from '@storybook/react';
 import type { CompositeStoreProps } from '../types';
 
 /**
- * Renders a composite widget.
+ * Renders a widget based on the WAI-ARIA [`composite`](https://w3c.github.io/aria/#composite) role,
+ * which provides a single tab stop on the page and arrow key navigation through the
+ * focusable descendants.
  *
  * ```jsx
  * const store = useCompositeStore();

```

Also a few docs nits I noticed (sorry to piggyback):

- The JSDoc/README snippets need to include the import statement (`import { Composite, useCompositeStore } from '@wordpress/components';`).
- The README contains a link to the Ariakit docs, which we should remove.

ciampo replied:
Good points! I went ahead, implemented your suggestions:

- [better composite description](https://github.com/WordPress/gutenberg/pull/64397/commits/dffa5bdd42ee5779945dbb9d580d6b30693030ab)
- [removed existing ariakit references from JSDocs and readmes](https://github.com/WordPress/gutenberg/pull/64397/commits/df64dc8461067148e29e76e57825fdbbcfa48055)
- [added imports in all code examples](https://github.com/WordPress/gutenberg/pull/64397/commits/87097490602ba200805e54ed64a6d036ad2f4192)

After that, I released that the prop description that came via Ariakit also referenced the ariakit website, other ariakit components / examples, or even other `Composite` props that we are not exposing. And therefore, I pushed a few more changes:

- instead of `Pick`-ing prop types, I added them explicitly (still re-using the original type) so that I could re-write the prop description in the JSDoc. In the process, I also updated Storybook and README with the updated JSDocs (which are not much more complete and closer to the original, but without ariakit references)(https://github.com/WordPress/gutenberg/pull/64397/commits/ad9fbb4147a9a2d55fc250525775538c88a4950b and https://github.com/WordPress/gutenberg/pull/64397/commits/929fb4c790ab91e16b4690418426a46879376fc3)
- For correctness, I added explicit default values in our implementation to make sure that our runtime code is true to the default values that we declare in the docs (https://github.com/WordPress/gutenberg/pull/64397/commits/32dd86843918a2860762ccd0eb0c2792cd9de8c4)

mirka reviewed (approved):
This is a really nice improvement! 🚀

tyxla reviewed (commented):
Thanks, @ciampo!

However, does it work as expected? When I navigate to the specific stories, the controls don't seem to work:


https://github.com/user-attachments/assets/9864f528-90fc-47f5-aaa4-4970fc7a0199

Am I missing something perhaps?

ciampo commented:
> Am I missing something perhaps?

@tyxla :

- The `orientation` prop changes the keyboard keys that are used to navigate across composite items (left/right for horizontal, up/down for vertical)
- The `rtl` prop inverts what prev and next do (so, for example, I'd expect the result of pressing the left/right arrow keys to be inverted when the prop is `true` and the orientation is `'horizontal'`)

They should not affect the component visually. That should be reflected in the props' description

ciampo committed:
Removed actions config in Storybook

ciampo committed:
Better composite description

tyxla commented:
🤦 of course, that makes a lot of sense!

Tests well and as expected ✅ Thanks!

ciampo committed:
Remove direct references to Ariakit's docs in JSDocs and README

ciampo committed:
Add import from `@wordpress/components` in all code snippets

ciampo committed:
useCompositeStore: update prop docs by using first-party docs

Instead of using Ariakit's definitions and descriptions, we use our own
version, which a copy or Ariakit's without any references to Ariakit, its
examples, or any other props that we don't expose.

ciampo committed:
useCompositeStore: set explicit default values

Along the same fashion as the previous commit, setting explicit default
values will give us more control when propagating ariakit changes, and
will allows to stay true to our first-part props docs.

ciampo committed:
Remove unnecessary space

ciampo committed:
Provide first-party prop descriptions also for other composite components

ciampo committed:
Add default value for store props to avoid errors while destructuring
